ipc/ipc/provider/src/lib.rs

```diff
@@ -479,6 +479,30 @@ impl IpcProvider {
             .await
     }
 
+    /// Funds an account in a child subnet with erc20 token, provided that the supply source kind is
+    /// `ERC20`. If `from` is None, it will use the default address config in `ipc.toml`.
+    /// If `to` is `None`, the `from` account will be funded.
+    pub async fn fund_with_token(
+        &mut self,
+        subnet: SubnetID,
+        from: Option<Address>,
+        to: Option<Address>,
+        amount: TokenAmount,
+    ) -> anyhow::Result<ChainEpoch> {
+        let parent = subnet.parent().ok_or_else(|| anyhow!("no parent found"))?;
```

Comment:
Does it make sense to perform a sanity check that the subnet indeed has a ERC20 supply source? If not, it will be caught anyway by the gateway actor, so probably not worth it. However, is this able to return the corresponding Solidity exception in a human-friendly form?

Reply:
yep, should be able to. From past experience, in contract execution, errors are usually returned.